### PR TITLE
single data transfer ldr word

### PR DESCRIPTION
**Pull request by guerinoni**
- Memory: Add unused memory as map
- Everywhere: Fix lint after bumping rust version
- ARM: Implement single data transfer LDR with WORD

### CODE REVIEW

File: ui/src/cpu_inspector.rs [hunk 1]
```diff
@@ -58,10 +58,10 @@ impl<T: Cpu> View for CpuInspector<T> {
         ui.heading("Registers");
         ui.add_space(8.0);
 
-        let registers = match self.gba.lock() {
-            Ok(gba) => gba.cpu.registers(),
-            Err(_) => vec![],
-        };
+        let registers = self
```
Comment (gallottino) [inline ui/src/cpu_inspector.rs:61]: Why this change? No problem about it, just for me it is more readable the first version. I don't know if there is some kind improvement.

File: src/main.rs [hunk 2]
```diff
@@ -7,16 +7,16 @@ fn main() {
...
     let args = env::args().skip(1).collect::<Vec<String>>();
 
-    let cartridge_name = match args.first() {
-        Some(name) => {
-            println!("loading {name}");
-            name.clone()
-        }
-        None => {
+    let cartridge_name = args.first().map_or_else(
+        || {
```
Comment (gallottino) [inline src/main.rs:11]: > Why this change? No problem about it, just for me it is more readable the first version. I don't know if there is some kind improvement

Same.

**Review by gallottino — APPROVED** [review]

File: ui/src/cpu_inspector.rs (hunk 1 above)
Comment (guerinoni) [reply ui/src/cpu_inspector.rs:61]: More idiomatic, and also clippy complains :D

File: src/main.rs (hunk 2 above)
Comment (guerinoni) [reply src/main.rs:11]: More idiomatic, and also clippy complains :D
